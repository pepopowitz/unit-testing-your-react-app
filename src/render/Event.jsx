import React from 'react';

import Location from './Location';
import Date from './Date';

export default function Event({ event }) {
    return (
        <article className="event">
            <h1>
                {event.name}
            </h1>

            <Date date={event.date} />
            <Location location={event.location} />
        </article>
    );
}
