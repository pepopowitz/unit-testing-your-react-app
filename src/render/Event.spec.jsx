import Event from './Event';
import React from 'react';

import { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';

it('renders an event', () => {
    const event = {
        name: 'Milwaukee Code Camp'
    };

    const wrapper = shallow(<Event event={event} />);

    expect(wrapper.find('.event')).toHaveLength(1);
});

it('doesn`t shallow render dates', () => {
    const event = {
        name: 'Milwaukee Code Camp',
        date: '2017-11-11'
    };

    const wrapper = shallow(<Event event={event} />);

    expect(wrapper.find('.event-date')).toHaveLength(0);
});

it('deep renders dates', () => {
    const event = {
        name: 'Milwaukee Code Camp',
        date: '2017-11-11'
    };

    const wrapper = mount(<Event event={event} />);

    expect(wrapper.find('.event-date')).toHaveLength(1);
});

it('snapshots an event shallowly', () => {
    const event = {
        name: 'Milwaukee Code Camp',
        date: '2017-11-11'
    };

    const wrapper = shallow(<Event event={event} />);

    expect(toJson(wrapper)).toMatchSnapshot();
});

it('snapshots an event deeply', () => {
    const event = {
        name: 'Milwaukee Code Camp',
        date: '2017-11-11'
    };

    const wrapper = mount(<Event event={event} />);

    expect(toJson(wrapper)).toMatchSnapshot();
});
