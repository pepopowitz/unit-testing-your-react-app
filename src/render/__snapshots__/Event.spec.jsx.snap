// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshots an event deeply 1`] = `
<Event
  event={
    Object {
      "date": "2017-11-11",
      "name": "Milwaukee Code",
    }
  }
>
  <article
    className="event"
  >
    <h1>
      Milwaukee Code
    </h1>
    <Date
      date="2017-11-11"
    >
      <h2
        className="event-date"
      >
        2017-11-11
      </h2>
    </Date>
    <Location />
  </article>
</Event>
`;

exports[`snapshots an event shallowly 1`] = `
<article
  className="event"
>
  <h1>
    Milwaukee Code Camp
  </h1>
  <Date
    date="2017-11-11"
  />
  <Location />
</article>
`;
